src/foam/nanos/dig/DigUtil.java

```diff
@@ -0,0 +1,84 @@
+/**
+ * @license
+ * Copyright 2019 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.nanos.dig;
+
+import foam.core.X;
+import foam.lib.*;
+import foam.lib.csv.CSVOutputter;
+import foam.lib.json.JSONParser;
+import foam.lib.json.Outputter;
+import foam.lib.json.OutputterMode;
+import foam.nanos.dig.exception.DigErrorMessage;
+import foam.nanos.http.Format;
+
+import javax.servlet.http.HttpServletResponse;
+import java.io.PrintWriter;
+
+public class DigUtil {
+  public static void outputException(X x, HttpServletResponse resp, Format format, PrintWriter out, DigErrorMessage error) {
+    if ( resp == null ) resp = x.get(HttpServletResponse.class);
```

Comment:
Why would resp be null? If it is, isn't that because of a bug somewhere else that we should fix?

Reply:
The null resp is for falling back to the response in the context so that it doesn't need to be passed in explicitly.

Reply:
But why wouldn't you pass it in explicitly? Either pass in the context or as an argument, but doing both doesn't make sense.

Reply:
Fixed in https://github.com/foam-framework/foam2/pull/2832/commits/54f8f146104f33c4fa1457457b6160af32926819.